### ISSUE TITLE
Pre hook that runs before content is saved

### ISSUE THREAD

**Issue by jenstornell** [getkirby-v2/panel#1116]
I thought there was an issue about it but it seems like it's not.

**Hook before save**

A hook that runs before save would be helpful in situations where the content needs to be manipulated before save.

If we use for example an update hook, it would write the file and then run the hook that will write to the file again. It will write to the file two times on each save.

**A real case is this issue**

https://github.com/Addpixel/KirbyComments/issues/35

In short, I want to flatten a nested structure before the content is saved.

Maybe it should be called `pre-hook` or perhaps `panel-filter`. I think panel filter is better.

**Comment by lukasbestle** [comment]
If I understand your use-case correctly, you want to overwrite the way the plugin stores custom fields by running your code after the plugin's code has been run. I'm not so sure about this as that's a hack.

**Comment by jenstornell** [comment]
- A filter (as I call it) should run the _before_ the content is saved.
- A hook runs _after_ the content is saved.

Every time we need to manipulate the data, there is no reason to save the file two times. Instead, it would be better if the manipulation would occur before the save, similar to a filter.

Hooks are good to use as triggers. Filters are good to use for preparing the data before the save.

**It would happen in this order:**

1. Click the Panel save.
1. Run filters.
1. Saves the content to file.
1. Run hooks.

But because there already are filters, maybe there is a better name for it.

**Comment by lukasbestle** [comment]
Makes sense. :)
